swatch-contracts/src/main/java/com/redhat/swatch/contract/model/ContractMapper.java

```diff
@@ -61,10 +63,19 @@ public interface ContractMapper {
   ContractMetricEntity metricDtoToMetricEntity(Metric metric);
 
   @Mapping(target = "subscriptionNumber", source = "contract.redHatSubscriptionNumber")
-  @Mapping(target = "vendorProductCode", source = "cloudIdentifiers.productCode")
+  @Mapping(target = "vendorProductCode", source = "cloudIdentifiers")
   @BeanMapping(ignoreByDefault = true)
   ContractEntity partnerContractToContractEntity(PartnerEntitlementContract contract);
 
+  default String extractVendorProductCode(PartnerEntitlementContractCloudIdentifiers code) {
```

Comment:
While this does work, it's relying a lot on rules of thumb on making the determination on who the cloud/billing provider is.  I think we should coordinate with the IT Partner Integration team (maybe as part of https://issues.redhat.com/browse/ITPART-1130, since it doesn't have a description yet?) to see if we can get a value added in the payload somewhere to explicitly tell us which cloud provider the contract is for.

Reply:
> so based on the structure I don't think we need a value to determine which cloud provider.

I disagree with this and think it'd be better if we could be explicit with which cloud provider we're working with instead of making assumptions based on structure.

Currently, we've only got aws or azure, and it's conveniently a 1:1 mapping.  AWS uses product code and  Azure uses Offer Id.  There's no overlap.  But knowing which key goes with which product is information that you would just kind of need to know because it's not obvious in the logic here...the way this code reads is "look for a product code first, if you don't find it, use offer id.  The words "product code" and "offer id" are pretty generic, and as more billing providers get added, it may not be appropriate to just say "use product code".  There could be a "product code" in the structure, but it not be what we want to map to our vendorProductCode.  This feels brittle right now.

Reply:
I see your point, i will reach out and ask them to add a field in the payload from UMB